### PR TITLE
Change the convention that versioneer uses

### LINKED ISSUE #216: ci: Update versioneer to use not including a "+" sign. Not support by GitHub
Since GitHub will change the "+" into a "." when package are uploaded, we can not use the "pep440-post" style.

Available options are documented here: https://github.com/warner/python-versioneer/blob/master/details.md#how-do-i-select-a-version-style

May be the `pep440`

### PR DESCRIPTION
**Pull request by dorukozturk**
## Motivation

Fixes #216.

### CODE REVIEW

File: src/pynwb/_version.py [hunk 1]
```diff
@@ -40,8 +40,8 @@ def get_config():
     # _version.py
     cfg = VersioneerConfig()
     cfg.VCS = "git"
-    cfg.style = "pep440-post"
-    cfg.tag_prefix = ""
+    cfg.style = "pep440-pre"
+    cfg.tag_prefix = "'*.*.*'"
```
Comment (jcfr) [inline src/pynwb/_version.py]: Is it expected to have single quote within double quote ?

Comment (dorukozturk) [reply src/pynwb/_version.py]: Nope my bad, fixing it right now.

File: setup.cfg [hunk 2]
```diff
@@ -2,7 +2,7 @@
 VCS = git
```
Comment (jcfr) [inline setup.cfg:2]: Suggested commit message:

```
packaging: Support download from GitHub release

This commit updates the convention used by versioneer
to pep440-pre [1]. Doing so allows to install the development packages
from the GitHub release. It ensures that no plus character
is included in the original package name. It was 
switched by a dot by GitHub and prevented successful install
of the package using:

  pip install pynwb --find-links https://github.com/NeurodataWithoutBorders/pynwb/releases/tag/latest  --no-index

[1] https://github.com/warner/python-versioneer/blob/master/details.md#how-do-i-select-a-version-style
```

**Comment by jcfr** [comment]
Let's not this hold up the release, we need to also update the `keep-pattern` here:

https://github.com/NeurodataWithoutBorders/pynwb/blob/80c26fac6150725f0542617b6e80a0c0d60bf6fc/.circleci/config.yml#L69

with:

```
--prerelease-packages-keep-pattern "*dev<DISTANCE>*" \
```

But before doing so, I need to update the mini-language of the publish script to support `<DISTANCE>`.

See http://scikit-ci-addons.readthedocs.io/en/latest/addons.html#mini-language-for-packages-selection

File: src/pynwb/_version.py (hunk 1 above)
Comment (dorukozturk) [reply src/pynwb/_version.py]: I take it back. It is autogenerated and expected.

**Pull request merged**